gpu-operator/release-notes.rst

```diff
@@ -33,6 +33,85 @@ Refer to the :ref:`GPU Operator Component Matrix` for a list of software compone
 
 ----
 
+.. _v25.10.0:
+
+25.10.0
+=======
+
+New Features
```

Comment:
I think another notable change to call out in the release notes is that we now configure containerd / cri-o using drop-in files by default. This impacts the install procedures for certain k8s distros, like microk8s and k0s, and users should be aware of this. Not sure if we should call this out as a new feature or as an improvement.

cc @tariq1890